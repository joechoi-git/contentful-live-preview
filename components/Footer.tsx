import React from "react";
import Link from "next/link";

export default function Footer() {
    return (
        <footer className="w-full text-center border-t border-slate-300 pt-12">
            <p>This is a sample app built to demonstrate the Contentful Live Preview feature.</p>
            <p>
                <Link
                    href="https://github.com/joechoi-git/contentful-live-preview"
                    target="_blank"
                    className="text-blue-500"
                >
                    Open GitHub repository.
                </Link>
            </p>
            <p className="pt-6">
                &copy; {new Date().getFullYear()}. All rights reserved by Joe Choi.
            </p>
        </footer>
    );
}
