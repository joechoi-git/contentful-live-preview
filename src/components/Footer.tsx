import React from "react";
import Link from "next/link";

export default function Footer() {
    return (
        <footer className="mt-6 w-full text-center border-t-2 border-secondary py-6">
            <p>This is a sample app built to demonstrate the Contentful Live Preview feature.</p>
            <p>
                <Link
                    href="https://github.com/joechoi-git/contentful-live-preview"
                    target="_blank"
                    className="text-primary"
                >
                    Open GitHub repository.
                </Link>
            </p>
            <p className="pt-6">&copy; {new Date().getFullYear()}</p>
        </footer>
    );
}
