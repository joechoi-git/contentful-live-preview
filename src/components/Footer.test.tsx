/* eslint-disable react/display-name */
import React from "react";
import { render, screen } from "@testing-library/react";
import Footer from "../components/Footer";

jest.mock("next/link", () => {
    return ({
        children,
        href,
        target,
        className
    }: {
        children: React.ReactNode;
        href: string;
        target?: string;
        className?: string;
    }) => (
        <a href={href} target={target} className={className}>
            {children}
        </a>
    );
});

describe("Footer Component", () => {
    test("renders correctly", () => {
        const { asFragment } = render(<Footer />);
        expect(
            screen.getByText(
                "This is a sample app designed to demonstrate the Contentful Live Preview SDK"
            )
        ).toBeInTheDocument();
        expect(screen.getByText("Open GitHub repository.")).toBeInTheDocument();
        expect(screen.getByText(`Â© ${new Date().getFullYear()}`)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    test("has correct link attributes", () => {
        render(<Footer />);
        const linkElement = screen.getByText("Open GitHub repository.");
        expect(linkElement).toHaveAttribute(
            "href",
            "https://github.com/joechoi-git/contentful-live-preview"
        );
        expect(linkElement).toHaveAttribute("target", "_blank");
        expect(linkElement).toHaveClass("text-accent");
    });
});
