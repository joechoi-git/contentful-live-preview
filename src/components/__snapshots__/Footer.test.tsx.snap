// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component renders correctly 1`] = `
<DocumentFragment>
  <footer
    class="mt-6 w-full text-center border-t-2 border-secondary py-6"
  >
    <p>
      This is a sample app built to demonstrate the Contentful Live Preview feature.
    </p>
    <p>
      <a
        class="text-primary"
        href="https://github.com/joechoi-git/contentful-live-preview"
        target="_blank"
      >
        Open GitHub repository.
      </a>
    </p>
    <p
      class="pt-6"
    >
      Â© 2024
    </p>
  </footer>
</DocumentFragment>
`;
